src/OpenLoco/Viewport.cpp

```diff
@@ -35,17 +46,282 @@ namespace OpenLoco::Ui
         paint(rt, screenToViewport(intersection));
     }
 
+    struct StationBorder
+    {
+        uint32_t left;
+        uint32_t right;
+        uint16_t width;
+    };
+    static constexpr std::array<StationBorder, 4> kZoomToStationBorder = {
+        StationBorder{
+            ImageIds::curved_border_left_medium,
+            ImageIds::curved_border_right_medium,
+            3,
+        },
+        StationBorder{
+            ImageIds::curved_border_left_medium,
+            ImageIds::curved_border_right_medium,
+            3,
+        },
+        StationBorder{
+            ImageIds::curved_border_left_small,
+            ImageIds::curved_border_right_small,
+            1,
+        },
+        StationBorder{
+            ImageIds::curved_border_left_small,
+            ImageIds::curved_border_right_small,
+            1,
+        },
+    };
+
+    static constexpr std::array<int16_t, 4> kZoomToStationFonts = {
+        Font::medium_bold,
+        Font::medium_bold,
+        Font::small,
+        Font::small,
+    };
+
+    // 0x0048DF4D, 0x0048E13B
+    static void drawStationName(Gfx::RenderTarget& unZoomedRt, const Station& station, uint8_t zoom, bool isHovered)
+    {
+        if (!station.labelFrame.contains(unZoomedRt.getDrawableRect(), zoom))
+        {
+            return;
+        }
+
+        auto& borderImages = kZoomToStationBorder[zoom];
+
+        const auto colour = Colours::getTranslucent([&station]() {
+            if (station.owner == CompanyId::null)
+            {
+                return Colour::grey;
+            }
+            else
+            {
+                return CompanyManager::getCompanyColour(station.owner);
+            }
+        }(),
+                                                    isHovered ? 0 : 1);
```

Comment:
What's happening here, formatting-wise? :joy:

Reply:
Yeah it all went a bit odd after i joined the two functions (hovered and not hovered). I'll split up the lambda so it's easier to read.